lib/mapper.js

```diff
@@ -187,35 +169,77 @@ exports.group = function(msg) {
  * http://docs.intercom.io/#Companies
  *
  * @param {Object} company
- * @return {Object}
+ * @return {Array}
  * @api private
  */
 
 function formatCompany(company){
-  if (is.string(company)) company = { name: company };
-
-  var ret = {};
-  ret.name = company.name;
+  var ret = [];
+  // Wrapping in array to for DRY code
+  if (is.object(company)) company = [company];
+
+  if (is.string(company)){
+    var payload = {
+      name: company,
+      company_id: hash(company) // Legacy
+    };
 
-  if (company.id) {
-    ret.company_id = company.id;
-    delete company.id;
-  } else if (company.name) {
-    ret.company_id = hash(company.name);
+    ret.push(payload);
+  } else if (is.array(company)){
+    each(function(company){
+      company = formatTraits(company);
+      var customTraits = removeDuplicateTraits(company);
+      var payload = reject({
+        name: company.name,
+        company_id: company.id || dot(company, 'company_id') || hash(company.name),
+        remote_created_at: dot(company, 'created') || dot(company, 'createdAt'),
+        monthly_spend: dot(company, 'monthlySpend'),
+        plan: company.plan,
+        custom_attributes: flatten(customTraits),
+        remove: company.remove
+      });
+
+      ret.push(payload);
+    }, company);
   }
 
-  delete company.name;
+  return ret;
+}
+
+/**
+ * Remove all redundant traits that's been mapped semantically already
+ *
+ * https://doc.intercom.io/api/#custom-attributes
+ *
+ * @param {Object} traits
+ * @return {Object}
+ * @api private
+ */
 
-  if (company !== undefined) ret.custom_attributes = company;
+function removeDuplicateTraits(traits){
+  var customTraits = formatTraits(traits);
+  var semanticTraits = [
+    'companies',
+    'company',
+    'createdAt',
+    'created_at',
```

Comment:
`objcase.del` will handle the casing of these, so we don't need to list multiple versions of the same field